include/envoy/server/overload_manager.h

```diff
@@ -21,6 +24,22 @@ enum class OverloadActionState {
  */
 typedef std::function<void(OverloadActionState)> OverloadActionCb;
 
+/**
+ * Thread-local cache of the current state of each configured overload action.
+ */
+class OverloadActionStateCache : public ThreadLocal::ThreadLocalObject {
+public:
+  bool isActive(const std::string& action) const {
```

Comment:
Since this is eventually going to be called on the data path on HCM, I guess I'm wonder out loud whether we want to do a string hash/compare on new stream? It's probably lost in the noise of the overall overhead of new stream creation. An alternative would be to establish a map from the action strings to opaque integer tags that HCM can obtain at instantiation. Seems like micro-optimization, just curious about your thoughts here.

Reply:
Good idea - changed to allow doing the map lookup at at initialization time.